deeplake/core/transform/transform.py

```diff
@@ -78,13 +79,16 @@ def eval(
                 Defaults to False.
             read_only_ok (bool): If ``True`` and output dataset is same as input dataset, the read-only check is skipped. This can be used to read data in parallel without making changes to underlying dataset.
                 Defaults to False.
+            checkpoint_interval (int): If > 0, the transform will be checkpointed with a commit every ``checkpoint_interval`` input samples to avoid restarting full transform due to intermitten failures. If <= 0, no checkpointing is done.
```

Comment:
@AbhinavTuli We should add that if the transform fails, we reset to the previously successful checkpoint